chains/evm/contracts/pools/USDC/USDCTokenPoolProxy.sol

```diff
@@ -0,0 +1,378 @@
+    }
+
+    PoolAddresses memory pools = s_pools;
+
+    address destinationPool;
+
+    // The order of the branches is based on the expected frequency of each mechanism being used, in order to save
+    // gas on every call.
+    if (mechanism == LockOrBurnMechanism.CCTP_V2) {
+      destinationPool = pools.cctpV2Pool;
+    } else if (mechanism == LockOrBurnMechanism.CCTP_V1) {
+      destinationPool = pools.cctpV1Pool;
+    } else if (mechanism == LockOrBurnMechanism.LOCK_RELEASE) {
+      destinationPool = s_lockReleasePools[lockOrBurnIn.remoteChainSelector];
+    }
+
+    // If the destination pool is the zero address, then no mechanism has been configured for the outgoing tokens
+    // and thus the remote chain is not supported and should revert.
+    if (destinationPool == address(0)) {
+      revert InvalidDestinationPool();
+    }
+
+    // Transfer the tokens to the proper child pool, as this contract is only a proxy and will not perform
+    // the lock/burn itself.
+    i_token.safeTransfer(destinationPool, lockOrBurnIn.amount);
+
+    return USDCTokenPool(destinationPool).lockOrBurn(lockOrBurnIn);
+  }
+
+  /// @inheritdoc IPoolV1
+  function isSupportedChain(
+    uint64 remoteChainSelector
+  ) external view returns (bool) {
+    // If the outgoing mechanism is not set for a chain, then the chain is not supported because there cannot be a lock
+    // or burn operation.
+    return s_lockOrBurnMechanism[remoteChainSelector] != LockOrBurnMechanism.INVALID_MECHANISM;
+  }
+
+  /// @inheritdoc IPoolV1
+  function isSupportedToken(
+    address token
+  ) external view returns (bool) {
+    return address(i_token) == token;
+  }
+
+  /// @notice Signals which version of the pool interface is supported
+  function supportsInterface(
+    bytes4 interfaceId
+  ) public pure override returns (bool) {
+    return interfaceId == type(IPoolV1).interfaceId || interfaceId == type(IERC165).interfaceId;
+  }
+
+  /// @inheritdoc IPoolV1
+  function releaseOrMint(
+    Pool.ReleaseOrMintInV1 calldata releaseOrMintIn
+  ) public virtual returns (Pool.ReleaseOrMintOutV1 memory) {
+    // Since this proxy does not inherit from the TokenPool contract, it must manually validate the caller as an offRamp.
+    if (!i_router.isOffRamp(releaseOrMintIn.remoteChainSelector, msg.sender)) {
+      revert Unauthorized();
+    }
+
+    // In previous versions of the USDC Token Pool, the sourcePoolData only contained two fields, a uint64 and uint32.
+    // For structs stored only in memory, the compiler assigns each field to its own 32-byte slot, instead of tightly
+    // packing like in storage. This means that a message originating from a previous version of the pool will have a
+    // sourcePoolData that is 64 bytes long, indicating an inflight message originating from a previous version of
+    // the USDC Token pool.
+    // This branch must come before a version check, because the first field would be a uint64 and thus if a version
```

Comment:
This is why you shouldn't use `0` as a tag ;) 

use `CCTPVersion` for the first byte, 3 bytes of randomness for the last few. That prevents issues like these and allows you to do cleaner routing with tags first

Reply:
If you do the above, can you please already create 2 tags for CCIPv2 and route them both to the v2 pool, so we have one to use for 1.7. Should save us an upgrade of this proxy contract.

Within the pool, we can choose the mechanism based on the tag, even though they both route to the same CCTPv2 pool

Reply:
https://github.com/smartcontractkit/chainlink-ccip/pull/1134/commits/b92364e8fc3156742b36a0cf0404043f7034d9f2#diff-510793de9f7a7a1b0dd14ec9abdc694fe11b9eb710c18b19be666c59bc459840R31